modules/gdscript/gdscript_editor.cpp

```diff
@@ -3443,6 +3444,55 @@ static void _find_call_arguments(GDScriptParser::CompletionContext &p_context, c
 	r_forced = r_result.size() > 0;
 }
 
+namespace {
+
+/**
+ * @brief Used as the `on_applied` callback for GDScript completions of category `COMPLETION_OVERRIDE_METHOD`.
+ *        This method will attempt to automatically apply the `@override` annotation when using completion to generate an override.
+ *        If the `@override` annotation is already present, nothing will occur.
+ *
+ * @param code_edit The code editor instance.
+ * @param caret     The caret that was used for this application of the completion.
+ */
+void _update_overrides_after_completion(CodeEdit *code_edit, int caret) {
+	int line = code_edit->get_caret_line(caret);
+	// Backtrack in the current line and upwards a few lines to search for `@override`. If we don't see it, apply it inline before the `func` keyword.
+	bool has_override_annot = false;
+	int lines_traversed = 0;
+	int func_line = -1; // Generally speaking we expect the line with `func` to be the current line, but for sanity we'll search for it anyway.
+
+	while (!has_override_annot && lines_traversed < 10 && line >= 0) {
+		String text = code_edit->get_line(line);
+
+		if (text.contains("func")) {
+			if (func_line == -1) {
+				func_line = line;
+			} else {
+				break; // We've hit another `func` decl. Anything in or above this line doesn't apply to this function anymore.
+			}
+		}
+
+		if (text.contains("@override")) {
+			has_override_annot = true;
+			break;
+		}
+
+		line--;
+		lines_traversed++;
+	}
+
+	// TODO: Review stage - I'd prefer to insert the `@override` on the line before `func`, but I'm not sure how to do that.
+	if (!has_override_annot && func_line != -1) {
+		String text = code_edit->get_line(func_line);
+		code_edit->set_line(func_line, "@override " + text);
+		int caret_col = code_edit->get_caret_column(caret);
+		code_edit->set_caret_line(func_line, true, true, 0, caret);
+		code_edit->set_caret_column(caret_col + strlen("@override "), false, caret);
+	}
```

Comment:
As this comment says, I would have preferred the autofill to generate
```gdscript
@override
func something() -> void:
```
rather than 
```gdscript
@override func something -> void:
```

But I'm not familiar enough with the APIs here to know how to do that smoothly.